### PR TITLE
Moves the grid/list toggle into the nav bar

### PR DESCRIPTION
**Pull request by colinta**
And some pretty good refactors, too (*I hope!*).

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -103,6 +95,7 @@ plugin 'cocoapods-keys', {
 post_install do |installer|
   installer.pods_project.targets.each do |target|
     target.build_configurations.each do |config|
+      config.build_settings['WARNING_CFLAGS'] = '$(inherited) -Wno-error=private-header' if target.name == 'FBSnapshotTestCase'
```
Comment (colinta) [inline Podfile]: In cocoapods 1.1+, only one `post_install` block is allowed.  Most of this was repetition anyway.

File: Sources/Controllers/BaseElloViewController.swift [hunk 2]
```diff
@@ -80,13 +80,6 @@ public class BaseElloViewController: UIViewController, ControllerThatMightHaveTh
 
 // MARK: Search
 public extension BaseElloViewController {
-    func addSearchButton() {
-        elloNavigationItem.rightBarButtonItem = UIBarButtonItem(
-            image: InterfaceImage.Search.normalImage, style: .Done,
-            target: self, action: #selector(searchButtonTapped)
-        )
-    }
```
Comment (colinta) [inline Sources/Controllers/BaseElloViewController.swift]: I "moved/refactored" this into a `UIBarButtonItem` helper method, which accepts a `BaseElloViewController` and wires the action itself.

File: Podfile (hunk 1 above)
Comment (steam) [inline Podfile]: 👍

File: Sources/Controllers/Profile/ProfileViewController.swift [hunk 3]
```diff
@@ -194,38 +188,39 @@ public final class ProfileViewController: StreamableViewController {
     }
 
     private func setupNavigationBar() {
-        assignRightButtons()
```
Comment (colinta) [inline Sources/Controllers/Profile/ProfileViewController.swift]: Oops, I meant to rename `setupNavigationBar` to `setupNavigationItems`.  Doing that now.

File: Sources/Controllers/Profile/ProfileViewController.swift [hunk 4]
```diff
@@ -194,38 +188,39 @@ public final class ProfileViewController: StreamableViewController {
...
-    }
-
-    func assignRightButtons() {
+        let backItem = UIBarButtonItem.backChevron(withController: self)
+        let gridListItem = UIBarButtonItem.gridListItem(delegate: streamViewController, isGridView: streamViewController.streamKind.isGridView)
+        let shareItem = UIBarButtonItem(image: .Share, target: self, action: #selector(ProfileViewController.sharePostTapped(_:)))
+        let moreActionsItem = UIBarButtonItem(image: .Dots, target: self, action: #selector(ProfileViewController.moreButtonTapped))
+        let isCurrentUser = userParam == currentUser?.id || userParam == "~\(currentUser)"
```
Comment (colinta) [inline Sources/Controllers/Profile/ProfileViewController.swift]: Not all these `UIBarButtonItem`s are used, but it looked better to create them all and assign according to the user/currentUser/share-enabled logic.

**Comment by colinta** [comment]
```Executed 2383 tests, with 0 failures (0 unexpected) in 264.847 (265.825) seconds```

**Pull request merged**